pkg/artifactory/resource_artifactory_user.go

```diff
@@ -175,6 +175,17 @@ func resourceUserCreate(ctx context.Context, d *schema.ResourceData, m interface
 		return diag.FromErr(err)
 	}
 
+	// Artifactory PUT call for creating user with groups attribute set to empty/null always sets groups to "readers".
```

Comment:
Let's include the PUT call is actually a bug. So what we are doing here is a workaround and should be removed once that bug is fixed.